System.IO.Streams/MemoryStream.cs

```diff
@@ -73,6 +73,24 @@ public MemoryStream(byte[] buffer)
             _expandable = false;
             _origin = 0;
             _isOpen = true;
+            _isWritable = true;
+        }
+
+        /// <summary>
+        /// Initializes a new non-resizable instance of the <see cref="MemoryStream"/> class based on the specified byte array with the <see cref="CanWrite"/> property set as specified.
+        /// </summary>
+        /// <param name="buffer">The array of unsigned bytes from which to create the current stream.</param>
+        /// <param name="isWritable">A bool indicating whether the stream should be writable</param>
+        /// <exception cref="ArgumentNullException"><paramref name="buffer"/> is <see langword="null"/>.</exception>
+        public MemoryStream(byte[] buffer, bool isWritable)
+        {
+            _buffer = buffer ?? throw new ArgumentNullException();
```

Comment:
better move what is in the existing constructor MemoryStream(byte[] buffer) here and call from MemoryStream(byte[] buffer) this new constructor with true

Reply:
Not sure why that did not occur to me in the moment, but corrected.